atlas/AtlasMl/atlasml/clients/weaviate.py

```
@@ -535,6 +535,64 @@ def update_property_by_id(
+            logger.info(
+                f"--- DELETED {deleted_count} OBJECTS MATCHING "
+                f"{property_name}={property_value} ---"
+            )
+            return deleted_count
+
+        except ValueError:
+            # Re-raise validation errors
+            raise
+        except WeaviateQueryError as e:
+            logger.error(f"❌ Weaviate query error deleting by property: {e}")
+            raise WeaviateOperationError(f"Failed to delete by property: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)